face-landmarks-detection/demo/index.js

```diff
@@ -0,0 +1,275 @@
+    returnTensors: false,
+    flipHorizontal: false,
+    predictIrises: state.predictIrises
+  });
+  ctx.drawImage(
+      video, 0, 0, videoWidth, videoHeight, 0, 0, canvas.width, canvas.height);
+
+  if (predictions.length > 0) {
+    predictions.forEach(prediction => {
+      const keypoints = prediction.scaledMesh;
+
+      if (state.triangulateMesh) {
+        ctx.strokeStyle = GREEN;
+        ctx.lineWidth = 0.5;
+
+        for (let i = 0; i < TRIANGULATION.length / 3; i++) {
+          const points = [
+            TRIANGULATION[i * 3], TRIANGULATION[i * 3 + 1],
+            TRIANGULATION[i * 3 + 2]
+          ].map(index => keypoints[index]);
+
+          drawPath(ctx, points, true);
+        }
+      } else {
+        ctx.fillStyle = GREEN;
+
+        for (let i = 0; i < NUM_KEYPOINTS; i++) {
+          const x = keypoints[i][0];
+          const y = keypoints[i][1];
+
+          ctx.beginPath();
+          ctx.arc(x, y, 1 /* radius */, 0, 2 * Math.PI);
+          ctx.fill();
+        }
+      }
+
+      if(keypoints.length > NUM_KEYPOINTS) {
+        ctx.strokeStyle = RED;
+        ctx.lineWidth = 1;
+
+        const leftCenter = keypoints[NUM_KEYPOINTS];
+        const leftDiameterY = distance(
+          keypoints[NUM_KEYPOINTS + 4],
+          keypoints[NUM_KEYPOINTS + 2]);
+        const leftDiameterX = distance(
+          keypoints[NUM_KEYPOINTS + 3],
+          keypoints[NUM_KEYPOINTS + 1]);
+
+        ctx.beginPath();
+        ctx.ellipse(leftCenter[0], leftCenter[1], leftDiameterX / 2, leftDiameterY / 2, 0, 0, 2 * Math.PI);
+        ctx.stroke();
+
+        if(keypoints.length > NUM_KEYPOINTS + NUM_IRIS_KEYPOINTS) {
+          const rightCenter = keypoints[NUM_KEYPOINTS + NUM_IRIS_KEYPOINTS];
+          const rightDiameterY = distance(
+            keypoints[NUM_KEYPOINTS + NUM_IRIS_KEYPOINTS + 2],
+            keypoints[NUM_KEYPOINTS + NUM_IRIS_KEYPOINTS + 4]);
+          const rightDiameterX = distance(
+            keypoints[NUM_KEYPOINTS + NUM_IRIS_KEYPOINTS + 3],
+            keypoints[NUM_KEYPOINTS + NUM_IRIS_KEYPOINTS + 1]);
+
+          ctx.beginPath();
+          ctx.ellipse(rightCenter[0], rightCenter[1], rightDiameterX / 2, rightDiameterY / 2, 0, 0, 2 * Math.PI);
+          ctx.stroke();
+        }
+      }
+    });
+
+    if (renderPointcloud && state.renderPointcloud && scatterGL != null) {
+      const pointsData = predictions.map(prediction => {
+        let scaledMesh = prediction.scaledMesh;
+        return scaledMesh.map(point => ([-point[0], -point[1], -point[2]]));
+      });
+
+      let flattenedPointsData = [];
+      for (let i = 0; i < pointsData.length; i++) {
+        flattenedPointsData = flattenedPointsData.concat(pointsData[i]);
+      }
+      const dataset = new ScatterGL.Dataset(flattenedPointsData);
+
+      if (!scatterGLHasInitialized) {
+        scatterGL.setPointColorer((i) => {
+          if(i >= NUM_KEYPOINTS) {
+            return RED;
+          }
+          return BLUE;
```

Comment:
Hi,
Thanks for this new cool model! I was testing it here https://observablehq.com/d/a4b086b09d93a22b

Considering the flatted array is always in order, you can do this to set the Iris colors for the scatter plot points for multiple faces. You can see it live [here](https://observablehq.com/d/a4b086b09d93a22b), it's just a detail but at least the faces won't show red.

~~~js
        if (i % (NUM_KEYPOINTS + NUM_IRIS_KEYPOINTS * 2) > NUM_KEYPOINTS) {
          return RED;
        }
        return BLACK;
~~~